### PR TITLE
Mongo uri

### PR DESCRIPTION
**Pull request by ric-evans**

### CODE REVIEW

File: file_catalog/mongo.py [hunk 1]
```diff
@@ -2,25 +2,35 @@
...
+    def __init__(self, host=None, port=None, authSource=None, username=None, password=None, uri=None):
+
+        if uri:
+            logger.info(f"MongoClient args: uri={uri}")
+            self.client = MongoClient(uri, authSource=authSource).file_catalog
+        else:
+            logger.info('MongoClient args: host=%s, port=%s, username=%s', host, port, username)
+            self.client = MongoClient(host=host, port=port,
+                                      authSource=authSource,
+                                      username=username, password=password).file_catalog
```
Comment (dsschult) [inline file_catalog/mongo.py:33]: I wonder what happens if you don't pass one of these variables in the environment.  What kind of error message does it give?  Probably not very informative for fixing the problem.

File: file_catalog/mongo.py [hunk 2]
```diff
@@ -2,25 +2,35 @@
...
 
         self.client = MongoClient(host=host, port=port,
                                   authSource=authSource,
```
Comment (blinkdog) [inline file_catalog/mongo.py:36]: Does the line override what was set in the if/else block above?

Comment (dsschult) [reply file_catalog/mongo.py:36]: It looks like it does.  That seems like a copy/paste error?

File: file_catalog/mongo.py (hunk 1 above)
Comment (ric-evans) [reply file_catalog/mongo.py:33]: I assume, the Mongo Client would fail to connect. How can I test this?

Comment (dsschult) [reply file_catalog/mongo.py:33]: I guess write a test missing an arg, that is expected to fail?

Comment (dsschult) [reply file_catalog/mongo.py:33]: I don't think this is a blocker, just a note that it produces poor error messages in this case.

**Pull request merged**